src/main/java/org/gradle/profiler/studio/plugin/StudioPluginInstaller.java

```diff
@@ -16,22 +17,21 @@
  */
 public class StudioPluginInstaller {
 
-    /**
-     * Note: if you change this constant, and you load plugins always from the same location,
-     * then you will have also a plugin installed before on the classpath, and it might override your new classes.
-     */
     private static final String PLUGIN_INSTALL_FOLDER_NAME = "gradle-profiler-studio-plugin";
 
     public void installPlugin(LaunchConfiguration configuration) {
         // Delete previous directory in case the plugin directory is reused
-        deletePluginDirectory(configuration.getStudioPluginsDir());
+        File pluginsDir = configuration.getStudioPluginsDir().toFile();
+        File pluginInstallDir = new File(pluginsDir, PLUGIN_INSTALL_FOLDER_NAME);
+        deletePluginDirectory(pluginInstallDir);
         installPluginToDirectory(
             configuration.getStudioPluginsDir(),
             configuration.getStudioPluginJars()
         );
+        registerDeletePluginShutdownHook(pluginInstallDir);
```

Comment:
A shutdown hook seems like a heavy-handed approach. Suppose you want to run the profiler embedded in a JVM that never exits, then this won't be called at all. Can we remove the directory more explicitly?

Reply:
Good point. Moved to the close method of the StudioGradleClient